src/block/block_ext.c

```diff
@@ -1000,7 +1013,7 @@ __block_append(
     if ((ext = el->last) != NULL && ext->off + ext->size == off)
         ext->size += size;
     else {
-        ext = __block_off_srch_last(el->off, astack);
```

Comment:
If we append, should we set `need_traverse` to `false`? Unless `el->last` is NULL but this can be checked in `__block_off_srch_last`.

Reply:
hi, etienneptl.

    //Here, the el ->last metadata pointer is used
    if ((ext = el->last) != NULL && ext->off + ext->size == off) ----------Here, the el ->last metadata pointer is used
        ext->size += size;
    else {
        //This must be traversed because the subsequent * asteck [i] needs to be obtained through traversal here
        ext = __block_off_srch_last(el, astack, true);      
        if (ext != NULL && ext->off + ext->size == off)
            ext->size += size;
        else {
            WT_RET(__wt_block_ext_alloc(session, &ext));
            ext->off = off;
            ext->size = size;

            for (i = 0; i < ext->depth; ++i)
                *astack[i] = ext;                ----------------------------------------------- here
            ++el->entries;
        }

        /* Update the cached end-of-list */
        el->last = ext;
    }